homeassistant/components/recorder/statistics.py

```diff
@@ -126,43 +126,27 @@ def get_start_time() -> datetime:
     return start
 
 
-def _get_metadata_ids(
-    hass: HomeAssistant, session: scoped_session, statistic_ids: list[str]
-) -> list[str]:
-    """Resolve metadata_id for a list of statistic_ids."""
-    baked_query = hass.data[STATISTICS_META_BAKERY](
-        lambda session: session.query(*QUERY_STATISTIC_META_ID)
-    )
-    baked_query += lambda q: q.filter(
-        StatisticsMeta.statistic_id.in_(bindparam("statistic_ids"))
-    )
-    result = execute(baked_query(session).params(statistic_ids=statistic_ids))
-
-    return [id for id, _ in result] if result else []
-
-
 def _update_or_add_metadata(
     hass: HomeAssistant,
     session: scoped_session,
     statistic_id: str,
     new_metadata: StatisticMetaData,
-) -> str:
+) -> int:
     """Get metadata_id for a statistic_id, add if it doesn't exist."""
     old_metadata_dict = _get_metadata(hass, session, [statistic_id], None)
     if not old_metadata_dict:
         unit = new_metadata["unit_of_measurement"]
         has_mean = new_metadata["has_mean"]
         has_sum = new_metadata["has_sum"]
-        session.add(
-            StatisticsMeta.from_meta(DOMAIN, statistic_id, unit, has_mean, has_sum)
-        )
-        metadata_ids = _get_metadata_ids(hass, session, [statistic_id])
+        meta = StatisticsMeta.from_meta(DOMAIN, statistic_id, unit, has_mean, has_sum)
+        session.add(meta)
+        session.flush()  # Flush to get the metadata id assigned
```

Comment:
Is it expected that a manual flush is needed here, or is something wrong with the table models?